pyvista/utilities/misc.py

```diff
@@ -63,4 +63,51 @@ def uses_egl() -> bool:
     return 'EGL' in ren_win_str or 'OSOpenGL' in ren_win_str
 
 
+def copy_vtk_array(array, deep=True):
+    """Create a deep or shallow copy a of vtk array.
+
+    Parametres
+    ----------
+    array : vtk.vtkDataArray or vtk.vtkAbstractArray
+        VTK array.
+
+    deep : bool, optional
+        When ``True``, create a deep copy of the array. When ``False``, returns
+        a shallow copy.
+
+    Returns
+    -------
+    vtk.vtkDataArray or vtk.vtkAbstractArray
+        Copy of the original VTK array.
+
+    Examples
+    --------
+    Perform a deep copy of a vtk array.
+
+    >>> import vtk
+    >>> import pyvista
+    >>> arr = vtk.vtkFloatArray()
+    >>> _ = arr.SetNumberOfValues(10)
+    >>> arr.SetValue(0, 1)
+    >>> arr_copy = pyvista.utilities.misc.copy_vtk_array(arr)
+    >>> arr_copy.GetValue(0)
+    1.0
+
+    """
+    if not isinstance(array, (_vtk.vtkDataArray, _vtk.vtkAbstractArray)):
+        raise TypeError(f"Invalid type {type(array)}.")
+
+    vtk_type = str(type(array)).split('.')[-1][:-2]
+    if vtk_type not in dir(_vtk):  # pragma: no cover
+        raise RuntimeError(f'Type {vtk_type} not supported.')
```

Comment:
This, along with the superfuous imports in `_vtk` look a bit sketchy. Any chance we can replace all this (up to and including the `getattr()` call) with
```py
new_array = type(array)()
```
or are there edge cases for which we need the specific list of types to be imported?

Reply:
This was done primarily for debugging and should have been left out.

Resolved in 129477c3c4867152eb1dfce0f50730aa61c3bc0c.